packages/types/src/http/httpHandlerInitialization.ts

```diff
@@ -34,6 +34,15 @@ export interface NodeHttpHandlerOptions {
    */
   requestTimeout?: number;
 
+  /**
+   * Delay before the NodeHttpHandler checks for socket exhaustion,
+   * and emits a warning if the active sockets and enqueued request count is greater than
+   * 2x the maxSockets count.
+   *
+   * Defaults to connectionTimeout + requestTimeout or 3000ms if those are not set.
+   */
+  socketAcquisitionWarningTimeout?: number;
```

Comment:
Same comment as before, can we mark this with `@internal`?

Reply:
Alternatives are to not let this timeout be configurable, or have a different opt-in to the warning. 

Since users are dealing with a mysterious process exit, having the log output without opt-in I think would be better for their debugging experience.